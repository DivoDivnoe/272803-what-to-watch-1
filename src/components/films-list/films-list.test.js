import React from 'react';
import {create} from 'react-test-renderer';
import FilmsList from './films-list.jsx';

const mock = {
  movies: [
    {
      title: `Fantastic Beasts`,
      image: ``,
      genre: `comedy`,
      preview: ``,
    },
    {
      title: `Major Payne`,
      image: ``,
      genre: `comedy`,
      preview: ``,
    },
  ],
  clickHandler: jest.fn(),
};

describe(`FilmCard component`, () => {
  it(`renders correctly`, () => {
    const {movies, clickHandler} = mock;

    const tree = create(
        <FilmsList
          movies={movies}
          clickHandler={clickHandler}
        />,
        {createNodeMock: (el) => {
          return el;
        }}).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
