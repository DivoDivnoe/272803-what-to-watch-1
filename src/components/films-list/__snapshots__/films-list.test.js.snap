// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard component renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={null}
  >
    <video
      height="175"
      poster=""
      width="280"
    >
      Your browser does not support the video tag.
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Fantastic Beasts
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={null}
  >
    <video
      height="175"
      poster=""
      width="280"
    >
      Your browser does not support the video tag.
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Major Payne
      </a>
    </h3>
  </article>
</div>
`;
